holoviews/plotting/bokeh/util.py

```diff
@@ -542,10 +543,10 @@ def date_to_integer(date):
     """
     if isinstance(date, np.datetime64):
         date = dt64_to_dt(date)
-    if pd and isinstance(date, pd.Timestamp):
-        dt_int = date.timestamp()*1000
-    elif isinstance(date, (dt.datetime, dt.date)):
-        dt_int = time.mktime(date.timetuple())*1000
+    elif pd and isinstance(date, pd.Timestamp):
+        date = date.to_pydatetime()
+    if hasattr(date, 'timetuple'):
+        dt_int = calendar.timegm(date.timetuple())*1000
```

Comment:
Seeing this makes me a little uneasy with respect to use with cftime types.  I think the built-in Python `calendar` module always assumes the Gregorian calendar is used; in cftime, different date types operate on different calendars, so the "seconds since 1970-01-01T00:00:00" value could be different for the same time tuple, depending on the calendar.

But perhaps there's good reason for doing things this way (I'm not familiar with how dates are handled in bokeh/holoviews)?

Reply:
Thanks, definitely need another look at this. Is there a simple way to convert the values to a Gregorian calendar?

Reply:
Ah, I see; this is appropriate if that is what you are looking to do.  

However, I'm not sure if this is the ideal approach in general; for instance this will cause there to be extra space (and potentially a tick label) in the place of February 29th, 2000 for a no leap calendar (even though in that calendar that date does not exist).  See [an example in a gist with this branch](https://nbviewer.jupyter.org/gist/spencerkclark/477ccac28db88c54004fd9a4c5f2d4fe).  There are other possible edge cases too (in a 360-day calendar some dates do not exist in a Gregorian calendar, like February 30th).

I think nc-time-axis handles these edge cases properly, because it does not rely on converting dates to the Gregorian calendar.  [See how it handles the example above in this gist](https://nbviewer.jupyter.org/gist/spencerkclark/79e790a6c59c6655c687fade3f00a370) (I've also included an example of how to plot with dates generated from `xarray.cftime_range`).

I understand if it would be a lot more work to solve this in general though.

Reply:
Thanks again, I'll look at nc_time_axis in a bit more detail, running your notebook is currently giving me an error though, which I had already filled here: https://github.com/SciTools/nc-time-axis/issues/40.

```
ValueError: invalid year provided in cftime.DatetimeNoLeap(0, 11, 27, 1, 12, 0, 4, 3, 331)
```

What version of nc_time_axis and cftime have? I **think** I have the latest version of both:

```
print(cftime.__version__)
print(nc_time_axis.__version__)
1.0.0
1.1.0
```

Reply:
Sure thing, thanks for the link to that issue.  In my notebook I was using cftime 1.0.2.1 and nc_time_axis 1.1.0.  I can reproduce your issue if I downgrade cftime to version 1.0.0.

Reply:
Thanks, I'll try upgrading.

Reply:
Okay, I think we'll likely have to push ahead with the subtly wrong behavior in bokeh, as a proper solution would require writing custom bokeh models to correctly format dates with different calendars. Would you expect a warning when plotting dates in a non-Gregorian calendar until we've got a proper fix?